### PR TITLE
feat(server): add users management api

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
* **#7092** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7092?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7082** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7082?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5602** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5602?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6995** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6995?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/core/user/resolver.ts [hunk 1]
```diff
@@ -137,3 +140,111 @@ export class UserResolver {
...
   }
 }
+
+@InputType()
+class ListUserInput {
+  @Field(() => Int, { nullable: true, defaultValue: 0 })
+  skip!: number;
+
+  @Field(() => Int, { nullable: true, defaultValue: 20 })
+  first!: number;
```
Comment (darkskygit) [inline packages/backend/server/src/core/user/resolver.ts:148]: unusal naming in pagination api? I recommend using `limit`

File: packages/backend/server/src/core/user/resolver.ts [hunk 2]
```diff
@@ -137,3 +140,111 @@ export class UserResolver {
...
+  ) {}
+
+  @Query(() => [UserType], {
+    description: 'List registered users',
+  })
+  async users(
+    @Args({ name: 'filter', type: () => ListUserInput }) input: ListUserInput
+  ): Promise<UserType[]> {
+    const users = await this.db.user.findMany({
+      select: { ...this.user.defaultUserSelect, password: true },
```
Comment (darkskygit) [inline packages/backend/server/src/core/user/resolver.ts:183]: I think it is not a good idea to allow viewing passwords in the admin panel
we can allow admin to change passwords, but they should not be allowed to view existing passwords.

File: packages/backend/server/src/core/user/resolver.ts (hunk 1 above)
Comment (forehalo) [reply packages/backend/server/src/core/user/resolver.ts:148]: https://graphql.org/learn/pagination/

File: packages/backend/server/src/core/user/resolver.ts [hunk 3]
```diff
@@ -137,3 +140,111 @@ export class UserResolver {
...
+    return users.map(sessionUser);
+  }
+
+  @Query(() => UserType, {
+    name: 'userById',
+    description: 'Get user by id',
+  })
+  async getUser(@Args('id') id: string) {
+    const user = await this.db.user.findUnique({
+      select: { ...this.user.defaultUserSelect, password: true },
```
Comment (darkskygit) [inline packages/backend/server/src/core/user/resolver.ts:197]: same as above

File: packages/backend/server/src/core/user/resolver.ts (hunk 1 above)
Comment (forehalo) [reply packages/backend/server/src/core/user/resolver.ts:148]: we didn't follow the best practice of pagination as graphql spec recommends, anyway

File: packages/backend/server/src/core/user/resolver.ts (hunk 2 above)
Comment (forehalo) [reply packages/backend/server/src/core/user/resolver.ts:183]: the password won't be returned, it's only selected to show `hasPassword` flag

**Review by darkskygit — APPROVED** [review]

**Pull request merged**